DEVELOPER_GUIDE.md

```diff
@@ -184,6 +186,25 @@ Additionally, it is possible to attach one debugger to the cluster JVM and anoth
 ./gradlew :integTest -Dtest.debug=1 -Dcluster.debug=1
 ```
 
+## Backwards Compatibility Testing
+
+The purpose of Backwards Compatibility Testing and different types of BWC tests are explained [here](https://github.com/opensearch-project/opensearch-plugins/blob/main/TESTING.md#backwards-compatibility-testing)
+
+Use these commands to run BWC tests for k-NN:
+1. Rolling upgrade tests: `./gradlew :qa:rolling-upgrade:testRollingUpgrade`
+2. Full restart upgrade tests: `./gradlew :qa:restart-upgrade:testAgainstNewCluster`
+3. `./gradlew :qa:bwcTestSuite` is used to run all the above bwc tests together.
+
+Use this command to run BWC tests for a given Backwards Compatibility Version:
+```
+./gradlew :qa:bwcTestSuite -Dbwc.version=2.9.0
+```
+Here, we are testing BWC Tests with BWC version of plugin as 2.9.0.
+
+### Adding new tests
+
+Before adding any new tests to Backward Compatibility Tests, we should be aware that the tests in BWC are not independent. While creating an index, a test cannot use the same index name if it is already used in other tests.
```

Comment:
Maybe add a note that we do support only full restart and rolling upgrade tests, and when it makes sense to add test of each of those types.

Reply:
We also support mixed cluster test. It is just I have combined it with rolling upgrade tests.

Reply:
I think we should state this, that mixed cluster tests are combined with upgrade, and also some basic guidance when each type of test is needed

Reply:
Basic guidance of when each type of test is needed is already been mentioned [here](https://github.com/opensearch-project/neural-search/pull/531/files#diff-d5555724c2ebca1b1f9be92a74c90e84c2dcbc76631148801024e24372b7edc6R191)

Reply:
The mixed tests are a part of rolling upgrade as 
1) testAgainstOneThirdUpgraded Cluster
2) testAgainstTwoThiredUpgraded Cluster
3) Then Rolling upgrade tests runs

and the task of rolling upgrade depends on testAgainstTwoThiredUpgraded. So it is already quite understanding. Mentioning it seperately will confuse the people reading it.

Reply:
To me that explanation in core is quite vague: `plugin functionalities work as expected after each node upgrade`. 
The whole purpose of adding a doc is explain the process and make it simpler for contributors. I'm trying to think from a new external contributor perspective, I would like to see something along these lines:
"When you're adding a new feature think of adding bwc test along with it. Ideally you need to add tests of all three types: mixed, upgrade, restart. With current design mixed and upgrade tests are combined and are part of the same test class (example). In some cases some test types are not possible (example), in such case try to do your best and add what's possible."